app/src/main/java/io/github/sheasmith/discountdetective/ui/screens/ShoppingListScreen.kt

```diff
@@ -108,10 +115,12 @@ fun ShoppingListScreen(
 @OptIn(ExperimentalFoundationApi::class)
 @Composable
 private fun ProductList(
-    grouped: Map<Pair<String, String>, List<Triple<RetailerProductInformation, StorePricingInformation, Int>>>,
+    grouped: Map<Pair<String, String>, List<Triple<RetailerProductInformation, StorePricingInformation, ShoppingListItem>>>,
     retailers: Map<String, Retailer>,
     viewModel: ShoppingListViewModel,
     innerPadding: PaddingValues,
+    onCheckedItem: (ShoppingListItem, Boolean) -> Unit,
```

Comment:
You're probably better off passing the viewmodel to this, rather than using callbacks, as you will have access to the shoppinglistitem when both of these callbacks are called (i.e. we generally only use callbacks when you only pass part of an object, but need to act based on the full object).